package/scripts/postinstall

```diff
@@ -43,20 +43,42 @@ then
 fi
 
 # create missing directories
-sudo mkdir -vp Cellar Frameworks etc include lib opt sbin share var/homebrew/linked
+mkdir -vp Cellar Frameworks etc include lib opt sbin share var/homebrew/linked
 
-# set permissions
+# get valid logged in user
 logged_in_user=$(echo "show State:/Users/ConsoleUser" | scutil | awk '/Name :/ { print $3 }')
-group=$(id -gn "${logged_in_user}")
+
+# optionally pass an environment variable to `installer` to define user
+if [[ -n "${HOMEBREW_PKG_USER:-}" ]]
+then
+  logged_in_user="${HOMEBREW_PKG_USER}"
+elif [[ "${logged_in_user}" =~ _mbsetupuser|loginwindow|root ]] || [[ -z "${logged_in_user}" ]]
+then
+  # get last created account UID over 500
+  logged_in_user=$(dscl . list /Users UniqueID | awk '$2 > 500 { print $1 }' | tail -n 1)
+  # no valid accounts exist yet, use root temporarily
+  if [[ -z "${logged_in_user}" ]]
+  then
+    logged_in_user="root"
+  fi
+fi
+
+# set permissions
 if [[ "${homebrew_directory}" == "/usr/local/Homebrew" ]]
 then
-  chown -R "${logged_in_user}:${group}" Cellar Frameworks Homebrew bin etc include lib sbin share var/homebrew/linked
+  chown -R "${logged_in_user}:admin" Cellar Frameworks Homebrew bin etc include lib sbin share var/homebrew/linked
 else
-  chown -R "${logged_in_user}:${group}" .
+  chown -R "${logged_in_user}:admin" .
 fi
 
 # move API cache to ~/Library/Caches/Homebrew
-user_api_cache_dir=~"${logged_in_user}"/Library/Caches/Homebrew/api
-mkdir -vp "${user_api_cache_dir}"
-mv -v "${homebrew_directory}/cache_api/"* "${user_api_cache_dir}"
-rm -vrf "${homebrew_directory}/cache_api"
+if [[ ! "${logged_in_user}" == "root" ]]
+then
+  user_home_dir=$(dscl . read /Users/"${logged_in_user}" NFSHomeDirectory | awk '{ print $2 }')
+  user_cache_dir="${user_home_dir}/Library/Caches/Homebrew"
+  user_api_cache_dir="${user_cache_dir}/api"
+  mkdir -vp "${user_api_cache_dir}"
+  mv -v "${homebrew_directory}/cache_api/"* "${user_api_cache_dir}"
+  chown -R "${logged_in_user}:staff" "${user_cache_dir}"
```

Comment:
You're using `staff` in some places and `admin` in others.

Reply:
As is the script gets a user's group, which is almost always `staff`, and then sets ownership for `/opt/homebrew` and `~/Library/Caches/Homebrew`. I won't call the behavior incorrect, but it doesn't mimic what the Homebrew install script does. My goal was to match permissions for a typical install. I did check my test machines to ensure this was accurate across Intel and Apple silicon. 

`/opt/homebrew` should be `$username:admin`, `~/Library/Caches/Homebrew` (like with everything in `~/Library/Caches` should be `$username:staff`.